src/axom/core/examples/core_containers.cpp

```diff
@@ -219,12 +219,17 @@ void demoArrayDevice()
   // _basic_array_device_implicit_end
 
   // _basic_array_device_explicit_start
-  axom::ArrayView<int, 1, axom::MemorySpace::Device> view_of_dynamic_array(
-    device_array_dynamic);
+  using DeviceArrayView = axom::ArrayView<int, 1, axom::MemorySpace::Device>;
```

Comment:
In general, I agree that it would be better to use the direct types, but the docs seemed somewhat difficult to read due to the long lines. I opted for a compromise with a type alias inside the snippet.

For reference, here's what it currently looks like in develop (before the change):
![image](https://user-images.githubusercontent.com/5626552/142469673-c1935e3d-027f-416a-924c-71f1f93eb12f.png)